p2p/src/peer.rs

```diff
@@ -73,27 +65,30 @@ impl fmt::Debug for Peer {
 }
 
 impl Peer {
-	// Only accept and connect can be externally used to build a peer
-	fn new(info: PeerInfo, adapter: Arc<dyn NetAdapter>) -> Peer {
-		Peer {
-			info,
-			state: Arc::new(RwLock::new(State::Connected)),
-			tracking_adapter: TrackingAdapter::new(adapter),
-			connection: None,
-		}
-	}
-
 	pub fn accept(
-		conn: &mut TcpStream,
+		mut conn: TcpStream,
 		capab: Capabilities,
 		total_difficulty: Difficulty,
 		hs: &Handshake,
 		adapter: Arc<dyn NetAdapter>,
 	) -> Result<Peer, Error> {
 		debug!("accept: handshaking from {:?}", conn.peer_addr());
-		let info = hs.accept(capab, total_difficulty, conn);
+		let info = hs.accept(capab, total_difficulty, &mut conn);
 		match info {
-			Ok(peer_info) => Ok(Peer::new(peer_info, adapter)),
+			Ok(info) => {
+				let state = Arc::new(RwLock::new(State::Connected));
```

Comment:
This chunk looks like a Peer constructor (same code here and in `connect`), can we extract it into `Peer::new` function or smth  like that? We need to pass `conn` and `adapter`, all details would be hidden there